changes.d/6067.fix.md

```diff
@@ -0,0 +1 @@
+Fixed very quick respawn after task removal.
```

Comment:
Another tricky terminology issue to sort out 😬  We don't typically (well, never) use "the workflow" to mean just the scheduler task pool, so what will "added back into the workflow" mean to users?

I'll change it to this for now, to avoid saying exactly what the task is "added back" to: 
```suggestion
Fixed a bug that sometimes allowed suicide-triggered or manually removed tasks to be added back later.
```